### DISCUSSION TITLE
uCNC with bltouch-height map probe alarm and moving z up

### DISCUSSION THREAD

**Discussion by MarcoB979** [martin2250/OpenCNCPilot#194] [Q&A]
I have installed uCNC on my mks DLC32 with a bltouch as a probe. Single probe with macro works (g38, reset alarm after probe touches, set z to zero and move up a little). 

When applying the height map feature i expected a similar behaviour. But when i run the height map, the bltouch goes down at the first point and triggers an alarm (which is supposed to happen i think), but the alarm is not reset. Next point bltouch is still in alarm state so z axis moves up a little. Same for all the next points. Would this be something related to uCNC, did I make a mistake or is there something in the software that can be configured?

**Comment by deHarro** [comment]
Hi Marco!
You are referencing **uCNC**...
Here we are at **OpenCNCPilot**.
Are you sure, you are asking in the correct discussion?

If you by mistake mentioned the wrong software and are actually talking about OpenCNCPilot, then you have to know, that the underlying GRBL has no support for standard BlTouch sensors, out of the box.
But you may have a look over [here](https://www.harald-sattler.de/html/3dtouch_sensor.htm) or [here](https://www.harald-sattler.de/html/3dtouch_interface.htm) (language is German, but at the lower end of each page there is a link to Google translate).

Harry

**Reply by MarcoB979** [reply]
Hi harry. The great thing about uCNC firmware is that it does support a bltouch that acts as a 'normal' cnc probe. But i am relatively new to cnc and also opencncpilot software. When i do a manual probe using the macro i described probing with the bltouch works as it should. When running a height map, the bltouch probe stays in alarm state after first touch. It could be that in the opencncpilot software there might be a setting on what the probe is expected to when probing the height map or that this is simply done by the g38.x command. That is my question in the end. Does the height map use g38.x when probing the height map.

I also contacted patiente, the creator of uCNC, to check the bltouch module in the firmware. It looks like the problem is firmware related, but maybe opencncpilot can/could be configured too

**Reply by deHarro** [reply]
Oh, I see... I was totally offroad in seeing uCNC as a software as OCP.
In truth, it is a replacement for GRBL. I missed that fact - Sorry!
Since I do not use uCNC, I'm done... 😊

**Reply by deHarro** [reply]
But reading again your description...
BlTouch has an inbuild LED displaying the internal state of the sensor (touched, ready for touching, error, ...).
Does the BlTouch show "ready for touching" **shortly after** the first touch event?
If not, uCNC misses the "lower pen" signal for the sensor to again go into "ready for touching" state.
I don't know the colors for the different states of the sensor, you have to look that up elsewhere.

**Reply by MarcoB979** [reply]
What happens is the bltouch puts the pin out (blue color), does the probe and gives alarm (red/blue blinking) and normally should release the alarm immediately and stow (red light). Then next point it starts over. 

Currently it is not resetting the alarm and was wondering if this could be configured in the opencncpilot software. 

At this moment it looks like a missing release alarm instruction in the uCNC software and patiente is looking into this. Apart from that i was wondering if there was some way for opencncpilot to configure what the probe needs to to in each probe motion cycle. But it looks like this should be configured in uCNC

**Reply by deHarro** [reply]
I'm with you, thinking it's lying in uCNC, as I don't know an adequate option for probing in OCP.
OCP only moves Z up and expects Alarm state going OFF on its own.

**Reply by MarcoB979** [reply]
So it is doing a z move up and then a g38.x only. Then indeed it has to be fixed/handled in the firmware itself. Working on this with the creator of uCNC allready. To be continued
